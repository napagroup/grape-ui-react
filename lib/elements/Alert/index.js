"use strict";

var _Object$keys = require("@babel/runtime-corejs3/core-js-stable/object/keys");

var _forEachInstanceProperty = require("@babel/runtime-corejs3/core-js-stable/instance/for-each");

var _context, _context2;

var _Object$defineProperty = require("@babel/runtime-corejs3/core-js-stable/object/define-property");

_Object$defineProperty(exports, "__esModule", {
  value: true
});

var _component = require("./component");

_forEachInstanceProperty(_context = _Object$keys(_component)).call(_context, function (key) {
  if (key === "default" || key === "__esModule") return;

  _Object$defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _component[key];
    }
  });
});

var _styled = require("./styled");

_forEachInstanceProperty(_context2 = _Object$keys(_styled)).call(_context2, function (key) {
  if (key === "default" || key === "__esModule") return;

  _Object$defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _styled[key];
    }
  });
});