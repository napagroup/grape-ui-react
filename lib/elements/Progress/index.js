"use strict";

var _Object$keys = require("@babel/runtime-corejs3/core-js-stable/object/keys");

var _forEachInstanceProperty = require("@babel/runtime-corejs3/core-js-stable/instance/for-each");

var _context, _context2;

var _Object$defineProperty = require("@babel/runtime-corejs3/core-js-stable/object/define-property");

_Object$defineProperty(exports, "__esModule", {
  value: true
});

var _getProgress = require("./getProgress");

_forEachInstanceProperty(_context = _Object$keys(_getProgress)).call(_context, function (key) {
  if (key === "default" || key === "__esModule") return;

  _Object$defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _getProgress[key];
    }
  });
});

var _styled = require("./styled");

_forEachInstanceProperty(_context2 = _Object$keys(_styled)).call(_context2, function (key) {
  if (key === "default" || key === "__esModule") return;

  _Object$defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _styled[key];
    }
  });
});