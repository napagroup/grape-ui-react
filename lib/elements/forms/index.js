"use strict";

var _Object$keys = require("@babel/runtime-corejs3/core-js-stable/object/keys");

var _forEachInstanceProperty = require("@babel/runtime-corejs3/core-js-stable/instance/for-each");

var _context, _context2, _context3, _context4, _context5, _context6;

var _Object$defineProperty = require("@babel/runtime-corejs3/core-js-stable/object/define-property");

_Object$defineProperty(exports, "__esModule", {
  value: true
});

var _CheckboxField = require("./CheckboxField");

_forEachInstanceProperty(_context = _Object$keys(_CheckboxField)).call(_context, function (key) {
  if (key === "default" || key === "__esModule") return;

  _Object$defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _CheckboxField[key];
    }
  });
});

var _DateField = require("./DateField");

_forEachInstanceProperty(_context2 = _Object$keys(_DateField)).call(_context2, function (key) {
  if (key === "default" || key === "__esModule") return;

  _Object$defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _DateField[key];
    }
  });
});

var _Form = require("./Form");

_forEachInstanceProperty(_context3 = _Object$keys(_Form)).call(_context3, function (key) {
  if (key === "default" || key === "__esModule") return;

  _Object$defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _Form[key];
    }
  });
});

var _RadioField = require("./RadioField");

_forEachInstanceProperty(_context4 = _Object$keys(_RadioField)).call(_context4, function (key) {
  if (key === "default" || key === "__esModule") return;

  _Object$defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _RadioField[key];
    }
  });
});

var _SelectField = require("./SelectField");

_forEachInstanceProperty(_context5 = _Object$keys(_SelectField)).call(_context5, function (key) {
  if (key === "default" || key === "__esModule") return;

  _Object$defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _SelectField[key];
    }
  });
});

var _TextField = require("./TextField");

_forEachInstanceProperty(_context6 = _Object$keys(_TextField)).call(_context6, function (key) {
  if (key === "default" || key === "__esModule") return;

  _Object$defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _TextField[key];
    }
  });
});