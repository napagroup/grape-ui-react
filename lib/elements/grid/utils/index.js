"use strict";

var _interopRequireDefault = require("@babel/runtime-corejs3/helpers/interopRequireDefault");

var _Object$defineProperty = require("@babel/runtime-corejs3/core-js-stable/object/define-property");

_Object$defineProperty(exports, "__esModule", {
  value: true
});

exports.defaultFlexBoxStylesPropsToTrim = void 0;

var _keys = _interopRequireDefault(require("@babel/runtime-corejs3/core-js-stable/object/keys"));

var _propTypes = _interopRequireDefault(require("@styled-system/prop-types"));

var _Progress = require("../../../elements/Progress");

const defaultFlexBoxStylesPropsToTrim = [...(0, _keys.default)(_propTypes.default.background), ...(0, _keys.default)(_propTypes.default.border), ...(0, _keys.default)(_propTypes.default.flexbox), ...(0, _keys.default)(_propTypes.default.grid), ...(0, _keys.default)(_propTypes.default.layout), ...(0, _keys.default)(_propTypes.default.position), ...(0, _keys.default)(_propTypes.default.shadow), ...(0, _keys.default)(_propTypes.default.space), ...(0, _keys.default)(_Progress.getProgressPropTypes), 'boxSizing'];
exports.defaultFlexBoxStylesPropsToTrim = defaultFlexBoxStylesPropsToTrim;