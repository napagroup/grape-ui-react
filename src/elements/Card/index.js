export * from './component';
export * from './styled';
