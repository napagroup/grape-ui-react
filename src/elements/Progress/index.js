export * from './getProgress';
export * from './styled';
