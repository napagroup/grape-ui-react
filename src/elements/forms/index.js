export * from './CheckboxField';
export * from './DateField';
export * from './Form';
export * from './SelectField';
export * from './TextField';
