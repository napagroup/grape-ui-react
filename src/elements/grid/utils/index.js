import propTypes from '@styled-system/prop-types';
import { getProgressPropTypes } from 'src/elements/Progress';

export const defaultFlexBoxStylesPropsToTrim = [
  ...Object.keys(propTypes.background),
  ...Object.keys(propTypes.border),
  ...Object.keys(propTypes.flexbox),
  ...Object.keys(propTypes.grid),
  ...Object.keys(propTypes.layout),
  ...Object.keys(propTypes.position),
  ...Object.keys(propTypes.shadow),
  ...Object.keys(propTypes.space),
  ...Object.keys(getProgressPropTypes),
  'boxSizing',
];
