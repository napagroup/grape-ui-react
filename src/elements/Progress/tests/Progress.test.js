import React from 'react';
import * as ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import { css, keyframes, ThemeProvider } from 'styled-components';
import { Flex } from 'src/elements/grid';
import { Progress } from '../index';

const customKeyframe = keyframes`
  0% {
    background-color: red;
    opacity: 0.7;
    width: 0%;
  }
  20% {
    background-color: orange;
    width: 10%;
  }
  29% {
    width: 40%;
  }
  30% {
    background-color: yellow;
    width: 50%;
  }
  50% {
    background-color: green;
    width: 70%;
  }
  70% {
    background-color: blue;
    width: 75%;
  }
  80% {
    width: 77%;
  }
  93% {
    width: 90%;
  }
  100% {
    background-color: purple;
    opacity: 1;
    width: 100%;
  }
`;

const customCircularKeyframe = props => keyframes`
  0% {
    stroke: rgba(0,0,0,0);
    stroke-dashoffset: ${props.circumference};
  }
  50% {
    stroke: #32cd32;
  }
  100% {
    stroke: rgba(0,0,0,0);
    stroke-dashoffset: ${-1 * props.circumference};
  }
`;
describe('Linear Progress - snapshots', () => {
  let container;
  beforeEach(() => {
    container = document.createElement('div');
  });
  it('should return an Indeterminate Linear Progress with default styling', () => {
    act(() => {
      ReactDOM.render(
        <ThemeProvider theme={{}}>
          <Progress />
        </ThemeProvider>,
        container
      );
    });
    expect(container).toMatchSnapshot();
  });
  it('should return a Determinate Linear Progress with the appropriate styling', () => {
    act(() => {
      ReactDOM.render(
        <ThemeProvider theme={{}}>
          <Progress
            diameter={20}
            indicatorColor="orange"
            isDeterminate
            trackColor="#e0ffff"
          />
        </ThemeProvider>,
        container
      );
    });
    expect(container).toMatchSnapshot();
  });
  it('should return a Linear Progress with custom animations and captions', () => {
    act(() => {
      ReactDOM.render(
        <ThemeProvider theme={{}}>
          <Progress
            animationDuration="5s"
            animationIterationCount={2}
            animationName={
              // styled components keyframes and css functions
              css`${customKeyframe}`
            }
            caption={(
              <Flex
                justifyContent="flex-end"
                py={2}
              >
                <Progress
                  animationDuration="5s"
                  diameter={20}
                  isDeterminate
                  loop={false}
                  progressType="circular"
                  strokeWidth={2}
                />
              </Flex>
            )}
            hideTrack
            loop={false}
          />
        </ThemeProvider>,
        container
      );
    });
    expect(container).toMatchSnapshot();
  });
  it('should return a Linear Progress with custom caption, value and total', () => {
    const value = 20;
    const total = 150;
    const caption = `${value}% of ${total}`;
    act(() => {
      ReactDOM.render(
        <ThemeProvider theme={{}}>
          <Progress
            caption={caption}
            total={total}
            value={value}
          />
        </ThemeProvider>,
        container
      );
    });
    expect(container).toMatchSnapshot();
  });
});
describe('Circular Progress - snapshots', () => {
  let container;
  beforeEach(() => {
    container = document.createElement('div');
  });
  it('should return an Indeterminate Circular Progress with default styling', () => {
    act(() => {
      ReactDOM.render(
        <ThemeProvider theme={{}}>
          <Progress
            progressType="circular"
          />
        </ThemeProvider>,
        container
      );
    });
    expect(container).toMatchSnapshot();
  });
  it('should return a Determinate Circular Progress with the appropriate styling', () => {
    act(() => {
      ReactDOM.render(
        <ThemeProvider theme={{}}>
          <Progress
            indicatorColor="red"
            isDeterminate
            progressType="circular"
            trackColor="white.dark"
          />
        </ThemeProvider>,
        container
      );
    });
    expect(container).toMatchSnapshot();
  });
  it('should return a Circular Progress with custom animations', () => {
    act(() => {
      ReactDOM.render(
        <ThemeProvider theme={{}}>
          <Progress
            animationDuration="5s"
            animationName={
              // styled components keyframes and css functions
              css`${customCircularKeyframe}`
            }
            progressType="circular"
          />
        </ThemeProvider>,
        container
      );
    });
    expect(container).toMatchSnapshot();
  });
  it('should return a Circular Progress with custom caption, value and total', () => {
    act(() => {
      ReactDOM.render(
        <ThemeProvider theme={{}}>
          <Progress
            caption="3/5"
            captionProps={{
              textAlign: 'center',
            }}
            progressType="circular"
            total={5}
            value={3}
          />
        </ThemeProvider>,
        container
      );
    });
    expect(container).toMatchSnapshot();
  });
});
