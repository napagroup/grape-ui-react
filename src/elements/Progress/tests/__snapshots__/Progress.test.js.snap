// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circular Progress - snapshots should return a Circular Progress with custom animations 1`] = `
.c2 {
  height: 24px;
  width: 24px;
  display: block;
  max-width: 100%;
  overflow: hidden;
  margin: auto;
}

.c3 {
  -webkit-backface-visibility: hidden;
  fill: none;
  stroke: hsl(0,0%,75%);
}

.c1 {
  -webkit-animation-name: fvtopB;
  animation-name: fvtopB;
  -webkit-animation-duration: 5s;
  animation-duration: 5s;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
}

.c4 {
  -webkit-backface-visibility: hidden;
  fill: none;
  stroke: hsl(312.8,67.8%,47.5%);
  -webkit-animation-name: iwSrvQ;
  animation-name: iwSrvQ;
  -webkit-animation-duration: 5s;
  animation-duration: 5s;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c0 {
  box-sizing: border-box;
}

<div>
  <div
    class="c0 c1"
  >
    <svg
      class="c2"
      display="block"
      height="24"
      overflow="hidden"
      width="24"
    >
      <circle
        circumference="62.83185307179586"
        class="c3"
        cx="12"
        cy="12"
        r="10"
        stroke-width="4"
      />
      <circle
        circumference="62.83185307179586"
        class="c4"
        cx="12"
        cy="12"
        r="10"
        stroke-dasharray="62.83185307179586"
        stroke-width="4"
      />
    </svg>
  </div>
</div>
`;

exports[`Circular Progress - snapshots should return a Circular Progress with custom caption, value and total 1`] = `
.c5 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 0.8rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: center;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin-bottom: 1rem;
  margin-top: 0;
}

.c2 {
  height: 24px;
  width: 24px;
  display: block;
  max-width: 100%;
  overflow: hidden;
  margin: auto;
}

.c3 {
  -webkit-backface-visibility: hidden;
  fill: none;
  stroke: hsl(0,0%,75%);
}

.c1 {
  -webkit-animation-name: fvtopB;
  animation-name: fvtopB;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

.c4 {
  -webkit-backface-visibility: hidden;
  fill: none;
  stroke: hsl(312.8,67.8%,47.5%);
  -webkit-animation-name: kEfyFd;
  animation-name: kEfyFd;
  stroke-dashoffset: 25.132741228718345px;
}

.c0 {
  box-sizing: border-box;
}

<div>
  <div
    class="c0 c1"
    transform="rotate(-90deg)"
  >
    <svg
      class="c2"
      display="block"
      height="24"
      overflow="hidden"
      width="24"
    >
      <circle
        circumference="62.83185307179586"
        class="c3"
        cx="12"
        cy="12"
        r="10"
        stroke-width="4"
      />
      <circle
        circumference="62.83185307179586"
        class="c4"
        cx="12"
        cy="12"
        r="10"
        stroke-dasharray="62.83185307179586"
        stroke-dashoffset="25.132741228718345px"
        stroke-width="4"
      />
    </svg>
  </div>
  <p
    class="c5"
  >
    3/5
  </p>
</div>
`;

exports[`Circular Progress - snapshots should return a Determinate Circular Progress with the appropriate styling 1`] = `
.c2 {
  height: 24px;
  width: 24px;
  display: block;
  max-width: 100%;
  overflow: hidden;
  margin: auto;
}

.c3 {
  -webkit-backface-visibility: hidden;
  fill: none;
  stroke: hsl(0,0%,90%);
}

.c1 {
  -webkit-animation-name: fvtopB;
  animation-name: fvtopB;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

.c4 {
  -webkit-backface-visibility: hidden;
  fill: none;
  stroke: hsl(4.1,89.6%,58.4%);
  -webkit-animation-name: wYSdJ;
  animation-name: wYSdJ;
  -webkit-animation-duration: 1.5s;
  animation-duration: 1.5s;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c0 {
  box-sizing: border-box;
}

<div>
  <div
    class="c0 c1"
    transform="rotate(-90deg)"
  >
    <svg
      class="c2"
      display="block"
      height="24"
      overflow="hidden"
      width="24"
    >
      <circle
        circumference="62.83185307179586"
        class="c3"
        cx="12"
        cy="12"
        r="10"
        stroke-width="4"
      />
      <circle
        circumference="62.83185307179586"
        class="c4"
        cx="12"
        cy="12"
        r="10"
        stroke-dasharray="62.83185307179586"
        stroke-width="4"
      />
    </svg>
  </div>
</div>
`;

exports[`Circular Progress - snapshots should return an Indeterminate Circular Progress with default styling 1`] = `
.c2 {
  height: 24px;
  width: 24px;
  display: block;
  max-width: 100%;
  overflow: hidden;
  margin: auto;
}

.c3 {
  -webkit-backface-visibility: hidden;
  fill: none;
  stroke: hsl(0,0%,75%);
}

.c1 {
  -webkit-animation-name: fvtopB;
  animation-name: fvtopB;
  -webkit-animation-duration: 1.5s;
  animation-duration: 1.5s;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
}

.c4 {
  -webkit-backface-visibility: hidden;
  fill: none;
  stroke: hsl(312.8,67.8%,47.5%);
  -webkit-animation-name: kEfyFd;
  animation-name: kEfyFd;
  -webkit-animation-duration: 1.5s;
  animation-duration: 1.5s;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c0 {
  box-sizing: border-box;
}

<div>
  <div
    class="c0 c1"
  >
    <svg
      class="c2"
      display="block"
      height="24"
      overflow="hidden"
      width="24"
    >
      <circle
        circumference="62.83185307179586"
        class="c3"
        cx="12"
        cy="12"
        r="10"
        stroke-width="4"
      />
      <circle
        circumference="62.83185307179586"
        class="c4"
        cx="12"
        cy="12"
        r="10"
        stroke-dasharray="62.83185307179586"
        stroke-width="4"
      />
    </svg>
  </div>
</div>
`;

exports[`Linear Progress - snapshots should return a Determinate Linear Progress with the appropriate styling 1`] = `
.c1 {
  background: #e0ffff;
}

.c3 {
  background: hsl(35.8,100%,50%);
  -webkit-animation-name: eudNeh;
  animation-name: eudNeh;
  -webkit-animation-duration: 1.5s;
  animation-duration: 1.5s;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c0 {
  box-sizing: border-box;
  height: 5px;
  overflow: hidden;
}

.c2 {
  box-sizing: border-box;
  height: 100%;
  width: 100%;
}

<div>
  <div
    class="c0 c1"
  >
    <div
      class="c2 c3"
    />
  </div>
</div>
`;

exports[`Linear Progress - snapshots should return a Linear Progress with custom animations and captions 1`] = `
.c7 {
  height: 20px;
  width: 20px;
  display: block;
  max-width: 100%;
  overflow: hidden;
  margin: auto;
}

.c8 {
  -webkit-backface-visibility: hidden;
  fill: none;
  stroke: hsl(0,0%,75%);
}

.c6 {
  -webkit-animation-name: fvtopB;
  animation-name: fvtopB;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

.c9 {
  -webkit-backface-visibility: hidden;
  fill: none;
  stroke: hsl(312.8,67.8%,47.5%);
  -webkit-animation-name: eFGfhQ;
  animation-name: eFGfhQ;
  -webkit-animation-duration: 5s;
  animation-duration: 5s;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c1 {
  background: transparent;
}

.c3 {
  background: hsl(312.8,67.8%,47.5%);
  -webkit-animation-name: lgmbxA;
  animation-name: lgmbxA;
  -webkit-animation-duration: 5s;
  animation-duration: 5s;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: 2;
  animation-iteration-count: 2;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c0 {
  box-sizing: border-box;
  height: 5px;
  overflow: hidden;
}

.c2 {
  box-sizing: border-box;
  height: 100%;
  width: 100%;
}

.c5 {
  box-sizing: border-box;
}

.c4 {
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 8px;
  padding-bottom: 8px;
}

<div>
  <div
    class="c0 c1"
  >
    <div
      class="c2 c3"
    />
  </div>
  <div
    class="c4"
  >
    <div
      class="c5 c6"
      transform="rotate(-90deg)"
    >
      <svg
        class="c7"
        display="block"
        height="20"
        overflow="hidden"
        width="20"
      >
        <circle
          circumference="56.548667764616276"
          class="c8"
          cx="10"
          cy="10"
          r="9"
          stroke-width="2"
        />
        <circle
          circumference="56.548667764616276"
          class="c9"
          cx="10"
          cy="10"
          r="9"
          stroke-dasharray="56.548667764616276"
          stroke-width="2"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Linear Progress - snapshots should return a Linear Progress with custom caption, value and total 1`] = `
.c4 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 0.8rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin-bottom: 1rem;
  margin-top: 0;
}

.c1 {
  background: hsl(0,0%,75%);
}

.c3 {
  background: hsl(312.8,67.8%,47.5%);
  -webkit-animation-name: hbshSv;
  animation-name: hbshSv;
}

.c0 {
  box-sizing: border-box;
  height: 5px;
  overflow: hidden;
}

.c2 {
  box-sizing: border-box;
  width: 13.333333333333334%;
  height: 100%;
}

<div>
  <div
    class="c0 c1"
  >
    <div
      class="c2 c3"
    />
  </div>
  <p
    class="c4"
  >
    20% of 150
  </p>
</div>
`;

exports[`Linear Progress - snapshots should return an Indeterminate Linear Progress with default styling 1`] = `
.c1 {
  background: hsl(0,0%,75%);
}

.c3 {
  background: hsl(312.8,67.8%,47.5%);
  -webkit-animation-name: hbshSv;
  animation-name: hbshSv;
  -webkit-animation-duration: 1.5s;
  animation-duration: 1.5s;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c0 {
  box-sizing: border-box;
  height: 5px;
  overflow: hidden;
}

.c2 {
  box-sizing: border-box;
  height: 100%;
  width: 100%;
}

<div>
  <div
    class="c0 c1"
  >
    <div
      class="c2 c3"
    />
  </div>
</div>
`;
